pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/MPSTNCuda.hpp

```diff
@@ -162,6 +168,35 @@ class MPSTNCuda final : public TNCudaBase<Precision, MPSTNCuda<Precision>> {
         setBasisState(zeroState);
     }
 
+    /**
+     * @brief Set the quantum state to the zero state.
+     */
+    void setZeroState() { reset(); }
+
+    /**
+     * @brief Update the ith MPS site.
+     *
+     * @param site_idx Index of the MPS site.
+     * @param host_data Pointer to the data on host.
+     * @param host_data_size Length of the data.
+     */
+    void updateIthMPSSite(const std::size_t site_idx, const ComplexT *host_data,
+                          std::size_t host_data_size) {
+        PL_ABORT_IF_NOT(
+            site_idx < BaseType::getNumQubits(),
+            "The site index should be less than the number of qubits.");
+
+        const std::size_t idx = BaseType::getNumQubits() - site_idx - 1;
+        PL_ABORT_IF_NOT(
+            host_data_size == tensors_[idx].getDataBuffer().getLength(),
```

Comment:
Do we want to abort or reallocate?

Reply:
Could you explain more about reallocate? It's just a safety check here.  This API update the initial quantum state with the host data provided by an user. Users should ensure the correctness of the data passed to this API and this is just a safety check.

Reply:
I'm wondering whether it would be possible to reallocate the data buffer with the correct length if needed.

Reply:
Unfortunately, no. This API just accepts user provided data (host-side) and copy it to the GPU device.